docs/src/rules/prefer-arrow-callback.md

```diff
@@ -110,6 +110,36 @@ someArray.map(function(item) { return this.doSomething(item); }, someObject);
 
 :::
 
+This rule additionally supports TypeScript type syntax.
```

Comment:
[Docs] Can this note the `this:` behavior, since that's the reason for the new functionality in this PR?

Reply:
Done!